### ISSUE TITLE
chainer.links.ResNetXXLayers fails to init

### ISSUE THREAD

**Issue by belltailjp** [chainer/chainer#3123]
I have found that `chainer.links.ResNetXXLayers` raises error when loading a caffemodel file.

### Reproduction environment
This environment is built from scratch using vagrant to confirm my error.
However I get the same error also in my actual environment.

* Ubuntu 16.04 x64 (vagrant)
* Python 3.6.2 (pyenv)
* Chainer 2.0.2
* cupy not used
* CUDA not used

### Reproduction procedure

First prepare caffemodel.

```
# download ResNet-xx-model.caffemodel from github:KaimingHe/deep-residual-networks
$ md5sum ResNet-50-model.caffemodel 
44b20660c5948391734036963e855dd2  ResNet-50-model.caffemodel
$ mv ResNet-50-model.caffemodel $HOME/.chainer/dataset/pfnet/chainer/models/ResNet-50-model.caffemodel
```

Then try to load the caffemodel via `chainer.links.ResNet50Layers`, you'll get the following error.

```
>> import chainer.links
>> chainer.links.ResNet50Layers()
Now loading caffemodel (usually it may take few minutes)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 331, in __init__
    super(ResNet50Layers, self).__init__(pretrained_model, 50)
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 112, in __init__
    pretrained_model, self)
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 717, in _retrieve
    lambda path: npz.load_npz(path, model))
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/dataset/download.py", line 151, in cache_or_load_file
    content = creator(temp_path)
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 716, in <lambda>
    path, lambda path: _make_npz(path, path_caffemodel, model, n_layers),
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 700, in _make_npz
    ResNet50Layers.convert_caffemodel_to_npz(path_caffemodel, path_npz, 50)
  File "/home/ubuntu/.pyenv/versions/3.6.2/lib/python3.6/site-packages/chainer/links/model/vision/resnet.py", line 147, in convert_caffemodel_to_npz
    chainermodel = cls(pretrained_model=None, n_layers=n_layers)
TypeError: __init__() got an unexpected keyword argument 'n_layers'
```

I got exactly the same error with both `ResNet101Layers` and `ResNet152Layers` as well as `ResNet50Layers`.

**Comment by niboshi** [comment]
I think this is fixed in #3088.
Can you test with latest `master`?

**Comment by belltailjp** [comment]
Oh, sorry I haven't checked the latest code. One day late.
I confirmed now it's working.
Thanks
